presto-resource-group-managers/src/main/java/com/facebook/presto/resourceGroups/db/DbResourceGroupConfigurationManager.java

```diff
@@ -253,6 +254,8 @@ private synchronized void populateFromDbHelper(Map<Long, ResourceGroupSpecBuilde
         List<SelectorSpec> selectors = dao.getSelectors()
                 .stream()
                 .filter(selectorRecord -> resourceGroupIdTemplateMap.containsKey(selectorRecord.getResourceGroupId()))
+                // sort the selectors in decreasing priority
+                .sorted(Comparator.<SelectorRecord>comparingLong(SelectorRecord::getPriority).reversed())
```

Comment:
I think you can static import `comparingLong`

Reply:
Actually, let's do the sort in the SQL. Then we don't need to grab the field at all and the entire change can be done in the DAO

Reply:
There's a filter happening before this, we can sort all entries, but then we are sorting more than necessary. If we are going on that route, I'd suggest adding a secondary index for priority.

Reply:
We should see if the filter can be pushed into the database